src/scripts/controllers/controller.ts

```diff
@@ -0,0 +1,46 @@
+import { View } from "../views/view";
+import { Services } from "../services/services";
+
+export class Controller {
+  private skipPosts: number;
+  private postsPerPage: number;
+  view: View;
+  apiServices: Services;
+
+  constructor(skipPosts: number = 0, postsPerPage: number = 4) {
+    this.skipPosts = skipPosts;
+    this.postsPerPage = postsPerPage;
+    this.apiServices = new Services();
+    this.view = new View();
+  }
+
+  async displayPost() {
+    try {
+      const data = await this.apiServices.getPost(
+        this.postsPerPage,
+        this.skipPosts
+      );
+
+      this.view.addPost(data);
+      this.skipPosts += this.postsPerPage;
+    } catch (error) {
+      if (error instanceof Error) {
+        alert(error.message);
+      } else {
+        alert("please try again later");
+      }
```

Comment:
You can use early returns to simplify this 👇 

```suggestion
      if (error instanceof Error) {
        return alert(error.message);
      }
      alert("please try again later");
```

Reply:
used early returns